source/adios2/core/IO.cpp

```diff
@@ -45,6 +45,8 @@ void IO::SetIOMode(const IOMode ioMode) { m_IOMode = ioMode; };
 
 void IO::SetParameters(const Params &parameters) { m_Parameters = parameters; }
 
+const Params &IO::GetParameters() { return m_Parameters; }
```

Comment:
It's not meaningless, it's part of the return type, which is const ref.  Which now that I think about it, the whole function should be const as well.

Which compilers does it warn on?